app/certificates.go

```diff
@@ -0,0 +1,441 @@
+package app
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/temporalio/tcld/utils"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	maxCADuration = 365 * 24 * time.Hour
+	minCADuration = 7 * 24 * time.Hour
+)
+
+func generateRandomString(n int) (string, error) {
+	const letters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-"
+	ret := make([]byte, n)
+	for i := 0; i < n; i++ {
+		num, err := rand.Int(rand.Reader, big.NewInt(int64(len(letters))))
+		if err != nil {
+			return "", err
+		}
+		ret[i] = letters[num.Int64()]
+	}
+
+	return string(ret), nil
+}
+
+type generateCACertificateInput struct {
+	Organization string        `validate:"required"`
+	Duration     time.Duration `validate:"required"`
+	RSAAlgorithm bool
+}
+
+func generateCACertificate(
+	input generateCACertificateInput,
+) (caPEM, caPrivateKeyPEM []byte, err error) {
+	validator := validator.New()
+	if err := validator.Struct(input); err != nil {
+		return nil, nil, err
+	}
+	randomLetters, err := generateRandomString(4)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	keyUsage := x509.KeyUsageDigitalSignature | x509.KeyUsageCRLSign | x509.KeyUsageCertSign
+	if input.RSAAlgorithm {
+		// Only RSA subject keys should have the KeyEncipherment KeyUsage bits set. In
+		// the context of TLS this KeyUsage is particular to RSA key exchange and
+		// authentication.
+		keyUsage |= x509.KeyUsageKeyEncipherment
+	}
+
+	dnsRoot := fmt.Sprintf("client.root.%s.%s", input.Organization, randomLetters)
+	now := time.Now()
+	conf := &x509.Certificate{
+		SerialNumber: big.NewInt(2019),
```

Comment:
How did we get 2019? Doesn't seem to follow spec: https://www.rfc-editor.org/rfc/rfc5280#section-4.1.2.2